universal-application-tool-0.0.1/app/controllers/AsyncController.java

```diff
@@ -0,0 +1,65 @@
+package controllers;
+
+import akka.actor.ActorSystem;
+import javax.inject.*;
+
+import akka.actor.Scheduler;
+import play.*;
+import play.mvc.*;
+import java.util.concurrent.Executor;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+
+import scala.concurrent.ExecutionContext;
+import scala.concurrent.duration.Duration;
+import scala.concurrent.ExecutionContextExecutor;
+
+/**
+ * This controller contains an action that demonstrates how to write
+ * simple asynchronous code in a controller. It uses a timer to
+ * asynchronously delay sending a response for 1 second.
+ */
+@Singleton
+public class AsyncController extends Controller {
```

Comment:
It'll be worth discussing as a team whether we want to implement all I/O async. It's certainly preferable from a throughput and scalability perspective, but I doubt the app will ever have enough traffic for that to be worth the tradeoff in development velocity.